### PR TITLE
Add support for Bidi_Paired_Bracket and Bidi_Paired_Bracket_Type.

### PR DESCRIPTION
**Pull request by deepakjois**
See #10 for more discussion on the motivation for adding this.

I do not see a huge change in binary size. The `a.out` binary generated from `ucdn-test.c` was 124K both before and after this change was made.

Please let me know if you would like any changes.

### CODE REVIEW

File: ucdn.c [hunk 1]
```diff
@@ -254,6 +266,34 @@ uint32_t ucdn_mirror(uint32_t code)
...
+
+    bp.from = code;
+    res = bsearch(&bp, bracket_pairs, BIDI_BRACKET_LEN, sizeof(BracketPair),
+            compare_bp);
+    if (res == NULL)
+        return UCDN_BIDI_PAIRED_BRACKET_TYPE_NONE;
+    else
+        return res->type;
+}
+
```
Comment (grigorig) [inline ucdn.c]: Please extract the code for finding the table entry into a separate function.

File: ucdn.h [hunk 2]
```diff
@@ -346,6 +350,25 @@ int ucdn_get_mirrored(uint32_t code);
...
+ * Get paired bracket type for a codepoint.
+ *
+ * @param code Unicode codepoint
+ * @return value according to UCDN_PAIRED_BRACKET_TYPE_* and as defined
+ * in UAX#9.
+ *
+ */
+int ucdn_paired_bracket_type(uint32_t code);
+
+/**
```
Comment (grigorig) [inline ucdn.h]: Also, I'm not sure about the API here. Maybe you need both the bracket type and thei paired bracket itself? Just a thought.

File: ucdn.h [hunk 3]
```diff
@@ -346,6 +350,25 @@ int ucdn_get_mirrored(uint32_t code);
...
+ * @return paired bracket codepoint or the original codepoint if no
+ * paired bracket character exists
+ */
+uint32_t ucdn_paired_bracket(uint32_t code);
+
+/**
+ * Get paired bracket type for a codepoint.
+ *
+ * @param code Unicode codepoint
+ * @return value according to UCDN_PAIRED_BRACKET_TYPE_* and as defined
```
Comment (grigorig) [inline ucdn.h]: The constants are named UCDN_BIDI_PAIRED_BRACKET_TYPE_*.

File: .gitignore [hunk 4]
```diff
@@ -0,0 +1,2 @@
+*.txt
+*.zip
```
Comment (grigorig) [inline .gitignore]: Please put this into a separate commit.

Comment (deepakjois) [inline .gitignore]: will fix

File: ucdn.c (hunk 1 above)
Comment (deepakjois) [inline ucdn.c]: will do.

File: ucdn.h (hunk 3 above)
Comment (deepakjois) [inline ucdn.h]: sorry about that. will do.

File: ucdn.h (hunk 2 above)
Comment (deepakjois) [inline ucdn.h]: Exactly, I modelled it on the mirroring API. The only minor difference is that we have the Bidi_Bracket_Pair_Type to help us distinguish between codepoints.

**Comment by deepakjois** [comment]
Addressed all comments on first commit.

**Comment by grigorig** [comment]
I just tested this with ucdn-test, and it doesn't seem to work, e.g. "ucdn-test 0x40" should report a paired bracket at 0x41/U+0041, right? Please check what's going on here.

**Comment by deepakjois** [comment]
U+0040 is LATIN CAPITAL LETTER A.

Here is an output for U+0028 LEFT PARENTHESIS (Which is 40 in decimal)

```
codepoint U+0028
combining_class 0
general_category 22
mirrored U+0029
paired bracket type open
paired bracket U+0029
eastasian_width 3
script 0
linebreak_class 0
bidi_class 18
```

Works for a few of the entries from the top of BidiBrackets.txt that I tested.

**Comment by grigorig** [comment]
OK, that is embarrassing - I actually confused decimal and hexadecimal notation. Sorry for that. Merged, thank you for the contribution.
